arch/armv7m/jmp.S

```diff
@@ -7,91 +7,118 @@
  * _setjmp, _longjmp, setjmp, sigsetjmp
  *
  * Copyright 2018, 2019 Phoenix Systems
- * Author: Kamil Amanowicz, Aleksander Kaminski
+ * Author: Kamil Amanowicz, Aleksander Kaminski, Jakub Sarzyński
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+ /* setjmp env buffer description
+  * Offset	Description
+  * 0		Signal mask flag - if != 0 then next value stores a signal mask
+  * 4		Signal mask
+  * 8		SP register
+  * 12-44	r4-r11 registers
+  * 44		LR register
+  * 48-80	d8-d15 registers
+  * 80		fpscr register
+  */
+
 #define __ASSEMBLY__
 
 .thumb
 .syntax unified
 
-.globl _setjmp
-.type _setjmp, %function
-_setjmp:
-	/* set jump address and stack pointer */
-	ldr r1, =(1f + 1)
-	mov r2, sp
-	/* store everythig in jmpbuf */
-	stm r0, {r1-r2, r4-r11, lr}
+
+.globl sigsetjmp
+.type sigsetjmp, %function
+sigsetjmp:
+	/* Store signal mask if savesigs != 0 */
+	mov r3, #0
+	sub r2, r1, #0
+	beq 1f
```

Comment:
Which statement sets the `Z` flag?
`subs r2, r1, #0`?

Reply:
Yes, missing s. I changed a implementation a little bit in favour of `cmp`.